src/plugins/redirections/crossing/checkCrossingRedirections.ml

```diff
@@ -0,0 +1,232 @@
+open Lintshell
+open Libmorbig.CST
+
+(* ================== [ Helpers about redirection lists ] =================== *)
+
+let cmd_prefix_to_io_redirect_list cmd_prefix' =
+  let rec aux acc = function
+    | CmdPrefix_IoRedirect io_redirect' ->
+       io_redirect' :: acc
+    | CmdPrefix_CmdPrefix_IoRedirect (cmd_prefix', io_redirect') ->
+       aux (io_redirect' :: acc) cmd_prefix'.value
+    | CmdPrefix_AssignmentWord _ ->
+       acc
+    | CmdPrefix_CmdPrefix_AssignmentWord (cmd_prefix', _) ->
+       aux acc cmd_prefix'.value
+  in
+  aux [] cmd_prefix'.value
+
+let cmd_suffix_to_io_redirect_list cmd_suffix' =
+  let rec aux acc = function
+    | CmdSuffix_IoRedirect io_redirect' ->
+       io_redirect' :: acc
+    | CmdSuffix_CmdSuffix_IoRedirect (cmd_suffix', io_redirect') ->
+       aux (io_redirect' :: acc) cmd_suffix'.value
+    | CmdSuffix_Word _ ->
+       acc
+    | CmdSuffix_CmdSuffix_Word (cmd_suffix', _) ->
+       aux acc cmd_suffix'.value
+  in
+  aux [] cmd_suffix'.value
+
+let redirect_list_to_io_redirect_list redirect_list' =
+  let rec aux acc = function
+    | RedirectList_IoRedirect io_redirect' ->
+       io_redirect' :: acc
+    | RedirectList_RedirectList_IoRedirect (redirect_list', io_redirect') ->
+       aux (io_redirect' :: acc) redirect_list'.value
+  in
+  aux [] redirect_list'.value
+
+(* ==================== [ Checker on redirection lists ] ==================== *)
+
+(* What we know on a descriptor *)
```

Comment:
I would have inserted a comment here to explain the principle of this quite involved analysis, and its limitations.

Reply:
Yes, this implementation lacks documentation. Also, it seems quite overkill when we could just be detecting patterns like `2>&1 >/dev/null` specifically. While we are talking about this, do you have a better name than "crossing"? Maybe something based on "leak"?

Reply:
What about "inverted"?

Reply:
I'll take that.